mesonbuild/environment.py

```diff
@@ -391,6 +399,7 @@ def __init__(self, source_dir, build_dir, options):
                 self.machines.host = MachineInfo.from_literal(config['host_machine'])
             if 'target_machine' in config:
                 self.machines.target = MachineInfo.from_literal(config['target_machine'])
+            self.paths.host.update(config.get('paths', {}))
 
         self.machines.default_missing()
         self.binaries.default_missing()
```

Comment:
N.B. would call `self.paths.default_missing()` with `PerMachineDefaultable` per the above.